### PR TITLE
working tests, solution, readme

### PR DESCRIPTION
**Pull request by maxwellbenton**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,18 +11,31 @@ to perform various operations on them.
 
 ## Perform Operations on Arrays
 
-We're going to write code to get all of the following results from the integers
-in our array:
+Your task is to write a series of methods that extract specific values from an
```
Comment (drakeltheryuujin) [inline README.md]: Is this language a little too technical for the audience? Might be worthwhile to note that integers are whole numbers and use different language (or expand on) "input array" so they don't get confused.

File: README.md [hunk 2]
```diff
@@ -11,18 +11,31 @@ to perform various operations on them.
...
-- whose cube is < 500
+```ruby
+array_of_integers = *0..50
+# => [0, 1, 2, 3, 4, 5, 6, 7, 8, 9,  ...  47, 48, 49]
+```
+
+> This method of array creation uses a 'splat' operator on a [range]. This is a
+> shorthand method for creating and assigning an array of integers
+
+Below this are the methods you must write implementation for. Read the provided
```
Comment (drakeltheryuujin) [inline README.md]: Is there something that should be below?

File: spec/array_spec.rb [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      expect(all_evens([1, 2, 3, 4, 5, 6])).to eq([2, 4, 6]), 'all_evens did not return the correct set of odd numbers'
+      expect(all_evens([1, 3])).to eq([]), 'all_evens should have returned an empty array if the input array is entirely odd integers'
+      expect(all_evens(array_of_integers_one)).to eq([0, 2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48]), 'all_evens did not return the correct set of odd numbers'
+      expect(all_evens(array_of_integers_two)).to eq([0, 2, 4, 6, 8]), 'all_evens did not return the correct set of odd numbers'
+      expect(all_evens([])).to eq([]), 'all_evens did not return an empty array when passed an empty array'
+    end
+  end
+
+  describe 'all_squares' do
+    it 'returns the square of all the numbers in array_of_integers' do
```
Comment (drakeltheryuujin) [inline spec/array_spec.rb]: I would specify "square root"

**Review by drakeltheryuujin — CHANGES_REQUESTED** [review]
Overall, this is a good lab. The only thing I can see making this a little too challenging is if students aren't brushed up on their old math formulas. If part of the assignment is to look up how to do each, that's fine, I would include in the instructions that they need an understanding of basic algebra/geometry(?). If not, define that squared numbers are the number multiplied by itself, cubed is the number multiplied by itself 3 times, etc.

File: lib/array_practice.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  # return all odd numbers in array_of_integers
+  array.select(&:odd?)
+end
+
+def all_evens(array)
+  # return all even numbers in array_of_integers
+  array.select(&:even?)
+end
+
+def all_squares(array)
```
Comment (drakeltheryuujin) [inline lib/array_practice.rb:13]: I would use "all_square_roots" and consistently use that language throughout

File: README.md (hunk 1 above)
Comment (maxwellbenton) [reply README.md]: agreed

File: lib/array_practice.rb (hunk 4 above)
Comment (maxwellbenton) [reply lib/array_practice.rb:13]: I've added better comments and some definitions in the readme to make this clearer.

File: README.md (hunk 2 above)
Comment (maxwellbenton) [reply README.md]: Aye, took this out

**Pull request merged**